its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/CppTest.java

```diff
@@ -45,28 +45,24 @@
 @ExtendWith({ServerTests.class, ContextExtension.class})
 class CppTest {
 
-  @TempDir
-  public Path basePath;
-
   @Test
   void testCppOnly() throws Exception {
-    String projectKey = "cpp";
-    String fileKey = "cpp:ConsoleApp/ConsoleApp.cpp";
+    var context = AnalysisContext.forServer("CppSolution");
+    String projectKey = context.projectKey;
+    String fileKey = projectKey + ":ConsoleApp/ConsoleApp.cpp";
 
-    ORCHESTRATOR.getServer().provisionProject(projectKey, "Cpp");
+    ORCHESTRATOR.getServer().provisionProject(projectKey, projectKey);
     ORCHESTRATOR.getServer().associateProjectToQualityProfile(projectKey, "cpp", QualityProfiles.CPP_S106);
 
-    Path projectDir = TestUtils.projectDir(basePath, "CppSolution");
+    Path projectDir = context.projectDir;
     File wrapperOutDir = new File(projectDir.toFile(), "out");
 
-    String token = TestUtils.getNewToken(ORCHESTRATOR);
-
-    TestUtils.newScannerBegin(ORCHESTRATOR, projectKey, projectDir, token)
+    context.begin
```

Comment:
The result of this execute call should be asserted to be successful. And adding the `.describeAs` will make it also more useful.